src/huggingface_hub/utils/_xet.py

```diff
@@ -12,75 +12,108 @@ class XetTokenType(str, Enum):
     """
-    # endpoint, access_token and expiration are required
     try:
         endpoint = headers[constants.HUGGINGFACE_HEADER_X_XET_ENDPOINT]
         access_token = headers[constants.HUGGINGFACE_HEADER_X_XET_ACCESS_TOKEN]
         expiration_unix_epoch = int(headers[constants.HUGGINGFACE_HEADER_X_XET_EXPIRATION])
     except (KeyError, ValueError, TypeError):
         return None
 
-    return XetMetadata(
+    return XetConnectionInfo(
         endpoint=endpoint,
         access_token=access_token,
         expiration_unix_epoch=expiration_unix_epoch,
-        refresh_route=headers.get(constants.HUGGINGFACE_HEADER_X_XET_REFRESH_ROUTE),
-        file_hash=headers.get(constants.HUGGINGFACE_HEADER_X_XET_HASH),
     )
 
 
 @validate_hf_hub_args
-def refresh_xet_metadata(
+def refresh_xet_connection_info(
     *,
-    xet_metadata: XetMetadata,
+    file_data: XetFileData,
     headers: Dict[str, str],
     endpoint: Optional[str] = None,
-) -> XetMetadata:
+) -> XetConnectionInfo:
     """
-    Utilizes the information in the parsed metadata to request the Hub xet access token.
+    Utilizes the information in the parsed metadata to request the Hub xet connection information.
+    This includes the access token, expiration, and XET service URL.
     Args:
-        xet_metadata: (`XetMetadata`):
-            The xet metadata provided by the Hub API.
+        file_data: (`XetFileData`):
+            The file data needed to refresh the xet connection information.
         headers (`Dict[str, str]`):
             Headers to use for the request, including authorization headers and user agent.
         endpoint (`str`, `optional`):
             The endpoint to use for the request. Defaults to the Hub endpoint.
     Returns:
-        `XetMetadata`: The metadata needed to make the request to the xet storage service.
+        `XetConnectionInfo`:
+            The connection information needed to make the request to the xet storage service.
     Raises:
         [`~utils.HfHubHTTPError`]
             If the Hub API returned an error.
         [`ValueError`](https://docs.python.org/3/library/exceptions.html#ValueError)
             If the Hub API response is improperly formatted.
     """
-    if xet_metadata.refresh_route is None:
+    if file_data.refresh_route is None:
         raise ValueError("The provided xet metadata does not contain a refresh endpoint.")
     endpoint = endpoint if endpoint is not None else constants.ENDPOINT
-    url = f"{endpoint}{xet_metadata.refresh_route}"
-    return _fetch_xet_metadata_with_url(url, headers)
+
+    # TODO: An upcoming version of hub will prepend the endpoint to the refresh route in
+    # the headers. Once that's deployed we can call fetch on the refresh route directly.
+    if file_data.refresh_route.startswith("/"):
+        url = f"{endpoint}{file_data.refresh_route}"
+    else:
+        url = f"{file_data.refresh_route}"
```

Comment:
```suggestion
    url = file_data.refresh_route
    if url.startswith("/"):
        url = f"{endpoint}{url}"
```
(slight preference for this code style -and once the TODO is done, we'll simply remove the `if: ...` statement)